Userland/Libraries/LibGfx/Rect.h

```diff
@@ -557,6 +557,24 @@ class Rect {
         return points;
     }
 
+    template<typename U = T>
+    [[nodiscard]] Gfx::Rect<U> interpolated_to(Gfx::Rect<T> const& to, float factor) const
+    {
+        VERIFY(factor >= 0.0f);
+        VERIFY(factor <= 1.0f);
+        if (factor == 0.0f)
+            return *this;
+        if (factor == 1.0f)
+            return to;
+        if (this == &to)
+            return *this;
+        auto interpolated_left = round_to<U>(mix<float>(x(), to.x(), factor));
+        auto interpolated_top = round_to<U>(mix<float>(y(), to.y(), factor));
+        auto interpolated_right = round_to<U>(mix<float>(right(), to.right(), factor));
+        auto interpolated_bottom = round_to<U>(mix<float>(bottom(), to.bottom(), factor));
+        return { interpolated_left, interpolated_top, interpolated_right - interpolated_left + 1, interpolated_bottom - interpolated_top + 1 };
```

Comment:
There's `Rect::from_two_points` for this, but it's missing the `+ 1`. Maybe we should audit all callers of `from_two_points`, add the `+ 1` there, and call `from_two_points` here? But doesn't have to be in this PR.

Reply:
Yeah have to think about this one... but yes this seems a bit too invasive to do in this PR.